addons/website_payment/controllers/portal.py

```diff
@@ -0,0 +1,156 @@
+                partner_details = {}
+
+            countries = request.env['res.country'].sudo().search([])
+            descriptions = request.httprequest.form.getlist('donation_descriptions')
+
+            donation_options = json.loads(donation_options) if donation_options else {}
+            donation_amounts = json.loads(donation_options.get('donationAmounts', '[]'))
+
+            rendering_context.update({
+                'is_donation': True,
+                'partner': partner_sudo,
+                'transaction_route': '/donation/transaction/%s' % donation_options.get('minimumAmount', 0),
+                'partner_details': partner_details,
+                'error': {},
+                'countries': countries,
+                'donation_options': donation_options,
+                'donation_amounts': donation_amounts,
+                'donation_descriptions': descriptions,
+            })
+        return rendering_context
+
+    def _get_payment_page_template_xmlid(self, **kwargs):
+        if kwargs.get('is_donation'):
+            return 'website_payment.donation_pay'
+        return super()._get_payment_page_template_xmlid(**kwargs)
+
+    @http.route(
+        '/donation/get_acquirer_fees', type='json', methods=['POST'], auth='public', website=True, sitemap=False,
+    )
+    def get_acquirer_fees(self, acquirer_ids=None, amount=None, currency_id=None, country_id=None):
+        # Cast numeric parameters as int or float and void them if their str value is malformed
+        # acquirer_ids = json.loads(acquirer_ids or '[]')
+        amount = self.cast_as_float(amount)
+        currency_id, country_id = tuple(map(
+            self.cast_as_int, (currency_id, country_id)
+        ))
+
+        acquirers_sudo = request.env['payment.acquirer'].sudo().browse(acquirer_ids)
+
+        # Make sure that the currency exists and is active
+        currency = request.env['res.currency'].browse(currency_id).exists()
+        if not currency or not currency.active:
+            raise werkzeug.exceptions.NotFound  # The currency must exist and be active
+
+        country = request.env['res.country'].browse(country_id)
+
+        # Compute the fees taken by acquirers supporting the feature
+        fees_by_acquirer = {
+            acq_sudo.id: acq_sudo._compute_fees(amount, currency, country)
+            for acq_sudo in acquirers_sudo.filtered('fees_active')
+        }
+        return fees_by_acquirer
+
+    @http.route(
+        '/donation/transaction/<minimum_amount>', type='json', methods=['POST'], auth='public', website=True, sitemap=False,
+    )
+    def donation_transaction(self, amount, currency_id, partner_id, access_token, minimum_amount=0, **kwargs):
+        if float(amount) < float(minimum_amount):
+            raise ValidationError(_('Donation amount must be at least %.2f.', float(minimum_amount)))
+        public_partner_id = request.env.ref('base.public_partner')
```

Comment:
not multi website, the public user/partner is not always the xml one

Reply:
Now using `website.user_id.partner_id`